python_toolbox/convert_to_logfile.py

```diff
@@ -0,0 +1,244 @@
+# check https://www.tanksandtemples.org/tutorial/ for further examples
+
+import sys
+import os
+import glob
+import numpy as np
+from numpy import matrix
+import read_model
+                 
+def write_SfM_log(T, i_map, filename):
+    with open(filename, 'w') as f:
+        ii=0
+        for t in T:
+            p = t.tolist()
+            f.write(' '.join(map(str, i_map[ii])) + '\n')
+            f.write('\n'.join(' '.join(map('{0:.12f}'.format, p[i])) for i in range(4)))
+            f.write('\n')
+            ii = ii + 1
+
+def quat2rotmat(qvec):
+	rotmat = np.array([1- 2 * qvec[2]**2 - 2 * qvec[3]**2, \
+	2 * qvec[1] * qvec[2] - 2 * qvec[0] * qvec[3], \
+	2 * qvec[3] * qvec[1] + 2 * qvec[0] * qvec[2], \
+	\
+	2 * qvec[1] * qvec[2] + 2 * qvec[0] * qvec[3], \
+	1 - 2 * qvec[1]**2 - 2 * qvec[3]**2, \
+	2 * qvec[2] * qvec[3] - 2 * qvec[0] * qvec[1], \
+	\
+	2 * qvec[3] * qvec[1] - 2 * qvec[0] * qvec[2], \
+	2 * qvec[2] * qvec[3] + 2 * qvec[0] * qvec[1], \
+	1 - 2 * qvec[1]**2 - 2 * qvec[2]**2])
+	rotmat = rotmat.reshape(3,3)
+	return rotmat
+
+# Usage: convert_COLMAP_to_log([COLMAP SfM file] [output log-filename] [folder of the input images] [image format])
+def convert_COLMAP_to_log(filename, logfile_out, input_images, formatp):
+	dirname = os.path.dirname(filename)
+	cameras, images, points3D = read_model.read_model(dirname,'.bin')
+	jpg_list = glob.glob(input_images+'/*.'+formatp)
+	jpg_list.sort()
+	nr_of_images = len(jpg_list)
+
+	R = []
+	t = []
+	T = []
+	i_map = []
+	TF = []
+	i_mapF = []
+
+	ii=0
+	for key, im in images.items():
+		qvec = im[1]
+		r = quat2rotmat(qvec)
+		t1 = im[2]
+		R.append(r)
+		t.append(t1)
+		w = np.zeros((4,4))
+		w[3,3] = 1
+		w[0:3,0:3]=r
+		w[0:3,3]=t1 
+		A = matrix(w)
+		T.append(A.I)
+		image_name = im[4]
+		matching = [i for i,s in enumerate(jpg_list) if image_name in s]
+		ii=im[0]
+		i_map.append(np.array([ii,matching[0],0.0],dtype='int'))	
+	ww=np.array(i_map).tolist()	
+	idm = np.identity(4)
+	# log file needs an entry for every input image, if image is not part of the SfM bundle it will be assigned to the identity matrix
+	for k in range(0,nr_of_images):
+		try:
+			bundler_id = [i for i, item in enumerate(ww) if k==item[1]][0] # find the id of view nr. k
+			i_mapF.append(np.array([k,k,0.0],dtype='int'))	
+			TF.append(T[bundler_id])
+		except:
+			i_mapF.append(np.array([k,-1,0.0],dtype='int'))	
+			TF.append(idm)		
+	write_SfM_log(TF, i_mapF, logfile_out)
+
+# USAGE: convert_MVE_to_log([MVE SfM file] [output log-filename] [folder containing the mve views])
+def convert_MVE_to_log(filename, logfile_out, input_images):
+	bundlefile = filename
+	lines = open(bundlefile).read().split('\n')
+	views_list = glob.glob(views_folder + '/*/')
+	views_list.sort()
+	nr_of_views = len(views_list)
+	nr_of_images = int(lines[1].split(' ')[0])
+
+	R = []
+	t = []
+	T = []
```

Comment:
t and T are confusing. Can you use more descriptive names for the variables? For example, translation and transformation